test/integration-test/src/utils.rs

```diff
@@ -0,0 +1,76 @@
+//! Utilities to run tests
+
+use std::{
+    fs::File,
+    io::Write,
+    os::fd::AsRawFd,
+    process::{Command, Stdio},
+    sync::atomic::{AtomicU64, Ordering},
+};
+
+use libc::{setns, CLONE_NEWNET};
+
+fn script(script: &str) {
+    // We pipe stdout/err to capture it and print it ourselves, so that it gets captured by the
+    // test harness
+    let mut bash = Command::new("/bin/bash")
+        .arg("-e")
+        .stdin(Stdio::piped())
+        .spawn()
+        .unwrap();
+
+    // Send script
+    let mut stdin = bash.stdin.take().unwrap();
+    stdin.write_all(script.as_bytes()).unwrap();
+    stdin.write_all(b"\n").unwrap();
+    drop(stdin);
+
+    // Wait for it to finish
+    let status = bash.wait().unwrap();
+
+    assert_eq!(status.code(), Some(0));
+}
+
+pub struct Netns {
+    pub name: String,
+    pub fd: File,
+}
+
+impl Netns {
+    pub fn new() -> Self {
+        static COUNTER: AtomicU64 = AtomicU64::new(0);
+        let name = format!("aya-test-{}", COUNTER.fetch_add(1, Ordering::Relaxed));
+        script(&format!(
+            "
+            ip netns add {name}
+            ip -netns {name} link set lo up
+        "
```

Comment:
Formatting here looks off

Reply:
sadly rustfmt won't let me do the following. fixed in 199d5ca

```rust
        script(&format!("
            ip netns add {name}
            ip -netns {name} link set lo up
        "));
```